MSAL/src/native_auth/public/MSALNativeAuthUserAccountResult.swift

```diff
@@ -70,30 +73,54 @@ import Foundation
+    /// Retrieves the access token for the default OIDC(openid, offline_access, profile) scopes from the cache.
     /// - Parameters:
     ///   - forceRefresh: Ignore any existing access token in the cache and force MSAL to get a new access token from the service.
     ///   - correlationId: Optional. UUID to correlate this request with the server for debugging.
     ///   - delegate: Delegate that receives callbacks for the Get Access Token flow.
-    @objc public func getAccessToken(forceRefresh: Bool = false, correlationId: UUID? = nil, delegate: CredentialsDelegate) {
-        Task {
-            let controllerResponse = await getAccessTokenInternal(
-                forceRefresh: forceRefresh,
-                correlationId: correlationId,
-                cacheAccessor: cacheAccessor
-            )
+    @objc public func getAccessToken(forceRefresh: Bool = false,
+                                     correlationId: UUID? = nil,
+                                     delegate: CredentialsDelegate) {
+        MSALLogger.log(
+            level: .info,
+            context: nil,
+            format: "Retrieving access token without scopes.")
 
-            let delegateDispatcher = CredentialsDelegateDispatcher(delegate: delegate, telemetryUpdate: controllerResponse.telemetryUpdate)
+        getAccessTokenInternal(forceRefresh: forceRefresh,
+                               scopes: [],
+                               correlationId: correlationId,
+                               delegate: delegate)
+    }
 
-            switch controllerResponse.result {
-            case .success(let accessTokenResult):
-                await delegateDispatcher.dispatchAccessTokenRetrieveCompleted(
-                    result: accessTokenResult,
-                    correlationId: controllerResponse.correlationId
-                )
-            case .failure(let error):
-                await delegate.onAccessTokenRetrieveError(error: error)
-            }
+    /// Retrieves the access token for the currently signed in account from the cache such that
+    /// the scope of retrieved access token is a superset of requested scopes. If the access token
+    /// has expired, it will be refreshed using the refresh token that's stored in the cache. If no
+    /// access token matching the requested scopes is found in cache then a new access token is fetched.
+    /// - Parameters:
+    ///   - forceRefresh: Ignore any existing access token in the cache and force MSAL to get a new access token from the service.
+    ///   - scopes: Permissions you want included in the access token received in the result. Not all scopes are guaranteed to be included in the access token returned.
+    ///   - correlationId: Optional. UUID to correlate this request with the server for debugging.
+    ///   - delegate: Delegate that receives callbacks for the Get Access Token flow.
+    public func getAccessToken(forceRefresh: Bool = false,
```

Comment:
I think default parameters should go after mandatory one. So, this should be
```suggestion
    public func getAccessToken(scopes: [String],
                                                     forceRefresh: Bool = false,
```

Reply:
Done.